Generierte Widgets schlecht mit CSS ansprechbar

Total-Reality opened the pull request:
Hallo,

ich möchte gerne anregen, dass die widgets zusätzliche DIVs bekommen sollten.

![widget_aenderung](https://cloud.githubusercontent.com/assets/5235894/11396783/8921652e-9374-11e5-8802-5d519ed205d4.jpg)

1) Siehe Screen Punkt 1.
Jedes widget eine zusätzliche Klasse bekommen und zwar sollte das der gleiche Name sein wie das Feld heißt.

Also wenn das Feld "company" heißt, sollte die widget-Klasse widget widget_company sein.
Ich habe mir jetzt anderweitig beholfen, Siehe Screen.

2) Siehe Screen Punkt 2 und 3
Das Label und das Input/Selectfeld etc. sollten mit einem zusätzlichen div umschlossen werden, damit man es global floaten kann. 

Ich glaube diese Art und Weise setzte früher auch die Erweiterung efg oder catalog ein.

Ich muss echt bei jedem Projekt mittlerweile ca. 10 Templates für jedes Widget anlegen, nur um das Floaten anständig hinzubekommen.

Hier mal ein Beispiel wie es dann final aussieht:
https://www.ihk-bildungsakademie-md.de/mein-profil/warenkorb/ihre-daten.html

leofeyer committed:
Merge branch 'hotfix/3.5.3' into develop

leofeyer committed:
Merge branch 'hotfix/3.5.4' into develop

leofeyer committed:
Merge branch 'hotfix/3.5.5' into develop

Toflar commented:
@Total-Reality Kannst du uns erklären was du genau mit deinen Pull Requests machst? Du willst vom `master` in den `develop` mergen, aber nicht von deinem Fork und Code ist auch nicht dabei?

Total-Reality commented:
Keine Ahnung, ich versteh das ganze System hier nicht. Ich will einfach nur n Ticket aufmachen und eine Änderung an den Widgets anregen. Was muss ich da tun?
Ich weiß auch nicht wie ich hier Code posten kann, steht nirgends ein Feld "Code" oder sowas...

Aybee commented:
Im Footer ist ein Help-Link https://help.github.com/
Dort dann nach **markdown** suchen.
https://help.github.com/articles/markdown-basics/
https://help.github.com/articles/github-flavored-markdown/

leofeyer committed:
Merge branch 'hotfix/3.5.6' into develop

Total-Reality commented:
Ohje... das ist echt super kompliziert... Ich hab schon über 20 Contao Erweiterungen programmiert, aber das hier ist mir echt zu hoch.

Warum Markdown? Ich will einfach nur n Ticket aufmachen :(

fiedsch commented:
```php
<?php 
echo "Damit Du Code lesbar in Deinem Ticket darstellen kannst!";
```

In Markdown


    ```php
    <?php 
    echo "Damit Du Code lesbar in Deinem Ticket darstellen kannst!";
    ```

Aybee commented:
> Warum Markdown? Ich will einfach nur n Ticket aufmachen :(

Das war die Antwort auf diese Frage:
> Ich weiß auch nicht wie ich hier Code posten kann, steht nirgends ein Feld "Code" oder sowas...

Total-Reality commented:
Danke, das probier ich mal. Soll ich das dann hier reinschreiben oder muss ich dann ein neues Ticket aufmachen?

Und kann mir noch jemand erklären wie Toflar das meint mit "Du willst vom master in den develop mergen, aber nicht von deinem Fork und Code ist auch nicht dabei?"

Was muss ich außerdem dem Code noch machen damit es richtig ist?

discordier commented:
Nicht auf Pull Request klicken sondern bei [issues](https://github.com/contao/core/issues) auf "[new issue](https://github.com/contao/core/issues/new)".

Pull requests ist dafyr, wenn man selber am code Dinge geaendert hat und diese als Aenderungswunsch einreichen moechte.
Hierzu forkt man das Projekt und schickt dann seine Aenderungen als PR an das Upstream projekt.
Siehe hierzu auch: 
- https://help.github.com/articles/fork-a-repo/
- https://help.github.com/articles/using-pull-requests/

Total-Reality commented:
Uff, ich hatte schon so einige Issues gestellt und wenn ich mich recht erinnere kam bei einem dieser Tickets von Leo Feyer oder wem anders die Aussage, dass mein Wunsch kein Fehler beinhalten würde, also kein Issue sein würde, sondern ein Pull Request...
Aber danke für eine endlich mal klare Aussage ;)

discordier commented:
Das ist ja auch korrekt, bei einem feature soll man einen PR stellen, aber MIT den Aenderungen darin.
Dann ist die Wahrscheinlichkeit dass es rein kommen kann groesser.
Wenn du des programmierens jedoch nicht maechtig bist, dann schreibst du ein normales Ticket und darin moeglichst detailliert was genau du moechtest (gerne auch mit screenshots wie in deinem Falle).
Dann jedoch muss irgendwann (wenn er eben die Zeit hat) einer hingehen und einen PR daraus machen wo dann Code drin ist. Das jedoch kann laaaaange dauern.

Total-Reality commented:
Wo schon erwähnt hab ich schon über 20 Erweiterungen programmiert, mittlerweile 30... 
Aber Github ist völlig schwachmatisch, hab jetzt in einem anderen Ticket die 3 accent grave gesetzt und danach php und als Folge wird der Code zwar gehighlightet, aber jetzt steht alles in einer Zeile ohne Zeilenumbruch!

Und wie gehen wir hier jetzt weiter vor? Soll ich ein neues Ticket erstellen oder den Ursprungsbeitrag oben mit dem Markdown-Code updaten?

leofeyer committed:
Merge branch 'hotfix/3.5.7' into develop

leofeyer committed:
Try to load the model from the registry before creating it.

mroi committed:
allow <picture> tag in custom HTML

leofeyer committed:
Merge branch 'hotfix/3.5.8' into develop

leofeyer committed:
Correctly filter the pages array in the quick link module.

leofeyer committed:
Fix the error 403/404 redirect (see contao/website#74).

bytehead committed:
Only treat size as an array, if it's not numeric

bytehead committed:
Remove unecessary unserialize call

bytehead committed:
Make sure that $size is always an array, even if $size is an integer

leofeyer committed:
Merge branch 'fix-addimagetotemplate' of https://github.com/bytehead/contao-core into hotfix/3.5.9

leofeyer committed:
Fix the phpDoc comment of the File class.

leofeyer committed:
Consider the fallback language if a page alias is ambiguous (see #8142).

leofeyer committed:
Correctly check if the request contains the numeric page ID.

discordier committed:
Correctly check error_reporting to fix issue #8254

We have to check the error_reporting() that has been set at runtime and
also walidate it against the passed error type. Simply checking for a
value greater than 0 is not enough.

Commit:
only add nofollow rel attribute to navigation links if the meta says 'noindex,nofollow'
So we allow spyders / bot to **crawl** sites  with 'noindex,follow' (but not to index)

leofeyer committed:
Handle the "float_left" and "float_right" classes in the back end (see #8239).

leofeyer committed:
Merge branch 'patch-1' of https://github.com/mroi/contao-core into hotfix/3.5.9

leofeyer committed:
Set the correct mime type when embedding SVG images (see #8245).

leofeyer committed:
Ensure that every image has a width and height attribute (see #8162).

leofeyer committed:
Correctly count the URLs when rebuilding the search index (see #8262).

leofeyer committed:
Allow to scroll the content of the SimpleModal iframe (see #8250).

leofeyer committed:
Show the 404 error page if an unpublished article is requested (see #8264).

leofeyer committed:
Merge branch 'fix-issue-8254' of https://github.com/discordier/core into hotfix/error-reporting

leofeyer committed:
Add a change log entry and a ticket reference.

leofeyer committed:
Merge branch 'hotfix/error-reporting' into hotfix/3.5.9

leofeyer committed:
Merge branch 'bugfix-nofollow_link_attribute' of https://github.com/DanielSchwiperich/core into hotfix/3.5.9

leofeyer committed:
Correctly handle custom mime icon paths (see #8275).

leofeyer committed:
Adjust the meta wizard field length to the column length (see #8277).

leofeyer committed:
Support visibility icons without link in sorting mode 5 (see #8242).

leofeyer committed:
Always use HTTPS if the target server supports SSL connections (see #8183).

leofeyer committed:
Correctly render the version diff link.

leofeyer committed:
Read from the temporary file if it has not been closed yet (see #8269).

leofeyer committed:
Respect the SSL settings of the root page when generating sitemaps (see #8270).

leofeyer committed:
Prevent the autofocus attribute from being added multiple times (see #8281).

leofeyer committed:
Fix issues found by the PhpStorm code inspector.

leofeyer committed:
Bump the version number.

leofeyer committed:
Merge branch 'hotfix/3.5.9' into develop
